org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/build/AbstractIncrementalBuilderTest.java

```diff
@@ -0,0 +1,182 @@
+	private OutputConfigurationProvider configurationProvider;
+
+	protected Multimap<URI, URI> generated;
+
+	protected List<URI> deleted;
+
+	protected List<Issue> issues;
+
+	protected InMemoryURIHandler inMemoryURIHandler;
+
+	@Before
+	public void setUp() {
+		clean();
+		inMemoryURIHandler = new InMemoryURIHandler();
+	}
+
+	protected List<Issue> clean() {
+		List<Issue> result = null;
+		generated = ArrayListMultimap.create();
+		deleted = new ArrayList<>();
+		result = issues = new ArrayList<>();
+		return result;
+	}
+
+	protected IndexState build(BuildRequest buildRequest) {
+		clean();
+		indexState = incrementalBuilder.build(buildRequest, (URI it) -> getLanguages().getResourceServiceProvider(it)).getIndexState();
+		return indexState;
+	}
+
+	protected void withOutputConfig(BuildRequest it, Procedure1<? super OutputConfiguration> init) {
+		OutputConfiguration config = Iterables.getFirst(configurationProvider.getOutputConfigurations(), null);
+		init.apply(config);
+		OutputConfigurationAdapter adapter = new OutputConfigurationAdapter(Collections.unmodifiableMap(
+				CollectionLiterals.newHashMap(Pair.of(languageName, Collections.unmodifiableSet(Sets.newHashSet(config))))));
+		it.getResourceSet().eAdapters().add(adapter);
+	}
+
+	protected abstract IResourceServiceProvider.Registry getLanguages();
+
+	protected BuildRequest newBuildRequest(Procedure1<? super BuildRequest> init) {
+		BuildRequest result = new BuildRequest();
+		ResourceDescriptionsData newIndex = indexState.getResourceDescriptions().copy();
+		result.setBaseDir(uri(""));
+		XtextResourceSet rs = resourceSetProvider.get();
+		rs.getURIConverter().getURIHandlers().clear();
+		rs.getURIConverter().getURIHandlers().add(inMemoryURIHandler);
+		rs.setClasspathURIContext(AbstractIncrementalBuilderTest.class.getClassLoader());
+		ProjectDescription projectDescription = new ProjectDescription();
+		projectDescription.setName("test-project");
+		projectDescription.attachToEmfObject(rs);
+		ChunkedResourceDescriptions index = new ChunkedResourceDescriptions(Collections.emptyMap(), rs);
+		index.setContainer(projectDescription.getName(), newIndex);
+		result.setResourceSet(rs);
+		result.setDirtyFiles(Collections.unmodifiableList(new ArrayList<>()));
+		result.setDeletedFiles(Collections.unmodifiableList(new ArrayList<>()));
+		result.setAfterValidate((URI uri, Iterable<Issue> issues) -> {
+			Iterables.<Issue> addAll(this.issues, issues);
+			return IterableExtensions.isEmpty(issues);
+		});
+		result.setAfterDeleteFile((URI uri) -> {
+			deleted.add(uri);
+		});
+		result.setAfterGenerateFile((URI source, URI target) -> {
+			generated.put(source, target);
+		});
+		result.setState(new IndexState(newIndex, indexState.getFileMappings().copy()));
+		init.apply(result);
+		return result;
+	}
+
+	protected URI delete(URI uri) {
+		try {
+			inMemoryURIHandler.delete(uri, Collections.emptyMap());
+			return uri;
+		} catch (IOException e) {
+			throw Exceptions.sneakyThrow(e);
```

Comment:
Use RuntimeException instead!?

Reply:
dont want to change the semantics on deprecated class